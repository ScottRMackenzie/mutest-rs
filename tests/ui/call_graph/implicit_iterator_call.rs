//@ print-targets
//@ stdout
//@ stderr: empty

fn next_impl() {}

struct ImplsIterator;

impl Iterator for ImplsIterator {
    type Item = ();

    fn next(&mut self) -> Option<Self::Item> {
        next_impl();
        None
    }
}

#[test]
fn test() {
    let iterator = ImplsIterator;
    for _ in iterator {}
}
