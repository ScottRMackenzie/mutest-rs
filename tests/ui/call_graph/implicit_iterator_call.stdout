tests -(1)-> next_impl at tests/ui/call_graph/implicit_iterator_call.rs:5:1: 5:15 (#0)
  (1) test

tests -(0)-> <ImplsIterator as std::iter::Iterator>::next at tests/ui/call_graph/implicit_iterator_call.rs:12:5: 12:45 (#0)
  (0) test

targets: 2 total; 2 safe; 0 unsafe (0 tainted)
